test/unit/kubernetes-deploy/kubernetes_resource/deployment_test.rb

```diff
@@ -2,118 +2,332 @@
 require 'test_helper'
 
 class DeploymentTest < KubernetesDeploy::TestCase
+  def setup
+    KubernetesDeploy::Kubectl.any_instance.expects(:run).never
+    super
+  end
+
   def test_deploy_succeeded_with_none_annotation
-    rollout = {
-      'metadata' => {
-        'name' => 'fake',
-        'annotations' => { KubernetesDeploy::Deployment::REQUIRED_ROLLOUT_ANNOTATION => 'none' }
-      }
+    deployment_status = {
+      "replicas" => 3,
+      "updatedReplicas" => 1,
+      "unavailableReplicas" => 1,
+      "availableReplicas" => 0
     }
 
-    deploy = KubernetesDeploy::Deployment.new(namespace: "", context: "", logger: logger, definition: rollout)
-    deploy.instance_variable_set(:@latest_rs, true)
-
+    rs_status = {
+      "replicas" => 3,
+      "availableReplicas" => 0,
+      "readyReplicas" => 0
+    }
+    dep_template = build_deployment_template(status: deployment_status, rollout: 'none',
+      strategy: 'RollingUpdate', max_unavailable: 1)
+    deploy = build_synced_deployment(template: dep_template, replica_sets: [build_rs_template(status: rs_status)])
```

Comment:
Should have made this more clear. I was thinking that `build_synced_deployment` would call `build_rs_template` & `build_deployment_template`. But this works if you like it better.

Reply:
I think this will be more flexible, in that a single test with esoteric needs could modify a field in the template before passing it into `build_synced_deployment`. This als makes it easy to write tests that need more than one replica set.